src/main/kotlin/com/atlassian/performance/tools/jirasoftwareactions/ActionShuffler.kt

```diff
@@ -0,0 +1,85 @@
+package com.atlassian.performance.tools.jirasoftwareactions
+
+import com.atlassian.performance.tools.jiraactions.api.SeededRandom
+import com.atlassian.performance.tools.jiraactions.api.WebJira
+import com.atlassian.performance.tools.jiraactions.api.action.Action
+import com.atlassian.performance.tools.jiraactions.api.action.BrowseProjectsAction
+import com.atlassian.performance.tools.jiraactions.api.action.CreateIssueAction
+import com.atlassian.performance.tools.jiraactions.api.action.SearchJqlAction
+import com.atlassian.performance.tools.jiraactions.api.measure.ActionMeter
+import com.atlassian.performance.tools.jiraactions.api.memories.IssueKeyMemory
+import com.atlassian.performance.tools.jiraactions.api.memories.JqlMemory
+import com.atlassian.performance.tools.jiraactions.api.page.IssuePage
+import com.atlassian.performance.tools.jiraactions.api.scenario.addMultiple
+import com.atlassian.performance.tools.jirasoftwareactions.api.actions.BrowseBoardsAction
+import kotlin.reflect.KClass
+
+class ActionShuffler {
+    private class FixedJqlMemory(val jql: String) : JqlMemory {
+        override fun observe(issuePage: IssuePage) {
+            throw UnsupportedOperationException()
+        }
+
+        override fun recall(): String? {
+            return jql
+        }
+
+        override fun remember(memories: Collection<String>) {
+            throw UnsupportedOperationException()
+        }
+    }
+
+    companion object {
+        fun createRandomisedScenario(seededRandom: SeededRandom, actionProportions: Map<Action, Int>,
+                                     issueKeyMemoriser: Action): List<Action> {
+            //createIssue needs a project - browserProject goes first
+            //viewIssue needs to have issues - createIssues goes second
+            //viewBoards needs to know about boards
```

Comment:
A unit test would be a better way to explain and protect such contracts.

BTW this function could be `internal` too.